### PR TITLE
Added flaky as a dependency for running tests

### PR DESCRIPTION
**Pull request by PritishSehzpaul**
**Context:** Flaky was missing as a required dependency from the doc. Without it `make test` fails.

**Description of the Change:** Added `flaky` as a dependency in the doc and why it is required.

**Benefits:** Better, more complete documentation.

**Possible Drawbacks:** None.

**Related GitHub Issues:** #1099

### CODE REVIEW

File: doc/development/guide/tests.rst [hunk 1]
```diff
@@ -5,11 +5,11 @@ Running the test suite
 ~~~~~~~~~~~~~~~~~~~~~~
 
 The PennyLane test suite requires the Python ``pytest`` package, as well as ``pytest-cov``
-for test coverage and ``pytest-mock`` for mocking; these can be installed via ``pip``:
+for test coverage, ``pytest-mock`` for mocking and ``flaky`` for automatically rerunning flaky tests; these can be installed via ``pip``:
```
Comment (josh146) [inline doc/development/guide/tests.rst:8]:
```suggestion
for test coverage, ``pytest-mock`` for mocking, and ``flaky`` for automatically rerunning flaky tests; these can be installed via ``pip``:
```

File: .github/CHANGELOG.md [hunk 2]
```diff
@@ -103,12 +103,13 @@
...
 - Typos addressed in templates documentation.
   [(#1094)](https://github.com/PennyLaneAI/pennylane/pull/1094)
+- Added `flaky` as dependency for running tests in documentation. [(#1113)](https://github.com/PennyLaneAI/pennylane/pull/1113)
 
 <h3>Contributors</h3>
 
 This release contains contributions from (in alphabetical order):
 
-Thomas Bromley, Kyle Godbey, Josh Izaac, Daniel Polatajko, Chase Roberts, Maria Schuld.
+Chase Roberts, Daniel Polatajko, Josh Izaac, Kyle Godbey, Maria Schuld, Pritish Sehzpaul, Thomas Bromley.
```
Comment (josh146) [inline .github/CHANGELOG.md:112]: @PritishSehzpaul we sort contributor names alphabetically by _last_ name, not first name! Could you revert the ordering here, and insert your name in the correct position.

File: .github/CHANGELOG.md [hunk 3]
```diff
@@ -103,12 +103,13 @@
 
 - Typos addressed in templates documentation.
   [(#1094)](https://github.com/PennyLaneAI/pennylane/pull/1094)
+- Added `flaky` as dependency for running tests in documentation. [(#1113)](https://github.com/PennyLaneAI/pennylane/pull/1113)
```
Comment (josh146) [inline .github/CHANGELOG.md:106]:
```suggestion

- Added `flaky` as dependency for running tests in documentation. [(#1113)](https://github.com/PennyLaneAI/pennylane/pull/1113)
```

**Review by josh146 — COMMENTED** [review]
Thanks @PritishSehzpaul for this contribution! Just left a comment regarding the contributor name ordering in the changelog, but apart from that it looks great and should be ready for merging directly after 🙂

File: .github/CHANGELOG.md (hunk 2 above)
Comment (PritishSehzpaul) [reply .github/CHANGELOG.md:176]: Gotcha. Thanks for letting me know. I will raise a commit

**Review by josh146 — APPROVED** [review]
Hi @PritishSehzpaul, a recent commit to master corrected the alphabetical ordering of contributors, so this should be good to merge in now. Thanks for the contribution!

**Pull request merged**